From 4ab62080f2c50b65c423a155cce8ab892d3e2e48 Mon Sep 17 00:00:00 2001
From: Steve Cornelius <steve.cornelius@freescale.com>
Date: Fri, 2 Aug 2013 14:19:27 -0700
Subject: [PATCH 0888/5242] MLK-9710-15 Correct size of padded key buffers

commit  d87a986977df286636d16595656affec340c5909 from
https://source.codeaurora.org/external/imx/linux-imx.git

Correct size of padded key buffers for the Secure Memory test module.

Signed-off-by: Steve Cornelius <steve.cornelius@freescale.com>
(cherry picked from commit da77cf5583c064deefd09a5e91851ab4398701a0)
Signed-off-by: Dan Douglass <dan.douglass@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/sm_test.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/sm_test.c b/drivers/crypto/caam/sm_test.c
index 8961fa2..841e300 100644
--- a/drivers/crypto/caam/sm_test.c
+++ b/drivers/crypto/caam/sm_test.c
@@ -91,8 +91,10 @@ int caam_sm_example_init(struct platform_device *pdev)
 	u32 unit, units;
 	int rtnval = 0;
 	u8 clrkey8[8], clrkey16[16], clrkey24[24], clrkey32[32];
-	u8 blkkey8[16], blkkey16[16], blkkey24[24], blkkey32[32];
-	u8 rstkey8[16], rstkey16[16], rstkey24[24], rstkey32[32];
+	u8 blkkey8[AES_BLOCK_PAD(8)], blkkey16[AES_BLOCK_PAD(16)];
+	u8 blkkey24[AES_BLOCK_PAD(24)], blkkey32[AES_BLOCK_PAD(32)];
+	u8 rstkey8[AES_BLOCK_PAD(8)], rstkey16[AES_BLOCK_PAD(16)];
+	u8 rstkey24[AES_BLOCK_PAD(24)], rstkey32[AES_BLOCK_PAD(32)];
 	u8 __iomem *blob8, *blob16, *blob24, *blob32;
 	u32 keyslot8, keyslot16, keyslot24, keyslot32 = 0;
 
-- 
1.7.9.5

